sycl/test/program_manager/env_vars.cpp

```diff
@@ -5,15 +5,12 @@
 //
 // RUN: %CPU_RUN_PLACEHOLDER SYCL_PROGRAM_COMPILE_OPTIONS="-g" %t.out
 // RUN: %GPU_RUN_PLACEHOLDER SYCL_PROGRAM_COMPILE_OPTIONS="-g" %t.out
-// RUN: %CPU_RUN_PLACEHOLDER SYCL_PROGRAM_LINK_OPTIONS="-enable-link-options -cl-denorms-are-zero" %t.out
-// RUN: %GPU_RUN_PLACEHOLDER SYCL_PROGRAM_LINK_OPTIONS="-enable-link-options -cl-denorms-are-zero" %t.out
```

Comment:
why is this removed?

Reply:
Hi, @kbobrovs 
Currently, all program's kernel code will be linked with our devicelib as default, so sycl runtime will use piProgramLink to link all spv images. However, after we enabled devicelib "on-demand" loading,  sycl runtime will use piProgramBuild to build users' device image. For OCL backend, piProgramBuild will be mapped to OCL's clBuildProgram API. According to OCL Spec, we can only pass "Compiler options" to this API and "Link Options" is not used.
To this test, the source code doesn't depend on devicelib, so I remove all test commands using "SYCL_PROGRAM_LINK_OPTIONS".
However, current status is a little tricky, we have  to keep loading "assert" fallback spv as default to workaround another separate issue, so sycl runtime will still use piProgramCompile + piProgramLink for all program's device code. Do you think we need to keep those test commands for now?
Thank you very much.

Reply:
I think the problems you describe are implementation detail and should be fixed. User setting `SYCL_PROGRAM_LINK_OPTIONS=` should not lead to crashes if link options are good. If that's the case, the implementation should be fixed, I think.

Reply:
Hi, @kbobrovs 
For this case, those link options will lead to crash in OCL CPU runtime, the problem is in "-enable-link-options". If no device libraries will be loaded and linked with user's kernel code, sycl runtime will use piProgramBuild for jit compilation which will be mapped to clBuildProgram. According to OCL Spec, "-enable-link-options" must be specified with "--create-library" option.  I think we have 2 ways to deal with the case:
1. If sycl runtime uses piProgramBuild, the SYCL_PROGRAM_LINK_OPTIONS env variable value will be ignored, we only pass SYCL_PROGRAM_COMPILER_OPTIONS value to underlying OCL runtime.
2. We keep passing "SYCL_PROGRAM_LINK_OPTIONS" to underlying OCL runtime even if piProgramBuild is used, it's user's duty to set "good" SYCL_PROGRAM_LINK_OPTIONS. We need to fix this case to remove "-enable-link-options".
Which one do you prefer or do you have any other idea?
Thank you very much.

Reply:
> According to OCL Spec, "-enable-link-options" must be specified with "--create-library" option.

I see, thanks. This makes the test invalid, so removal is OK then.
2) seems better solution. But for this PR just removal of the two invalid test cases (as you did before) should be enough